modules/core/src/v2/coins/xlm.ts

```diff
@@ -540,10 +602,9 @@ export class Xlm extends BaseCoin {
 
     const addressDetails = this.getAddressDetails(address);
     const rootAddressDetails = this.getAddressDetails(rootAddress);
-
-    if (addressDetails.address !== rootAddressDetails.address) {
+    if (addressDetails.baseAddress !== rootAddressDetails.address) {
```

Comment:
should this be `addressDetails.baseAddress !== rootAddressDetails.baseAddress` ?

Reply:
we don't need to, since `rootAddress` is passed as param and it's always the root (or base) address of the wallet